demo/MvvmNanoDemo/ViewModels/LoginViewModel.cs

```diff
@@ -43,12 +45,16 @@ public MvvmNanoCommand ShowAboutCommand
             get { return new MvvmNanoCommand(NavigateTo<AboutViewModel>); }
         }
 
-        private async void LogIn()
+        private void LogIn()
         {
             if (!IsFormValid)
                 return;
 
-            await NavigateToAsync<WelcomeViewModel, User>(new User(Username));
+            var application = Application.Current as MvvmNanoApplication;
+             
+            application.SetUpMasterDetailPage<MasterViewModel>();
+
+            ((MasterViewModel) application.MasterPage.BindingContext).Username = Username;  
```

Comment:
A view model may _never ever_ access the Application. This is only possible here because everything is hosted within the same lib.